hell/adventcalendar/2024/6/index.md

```diff
@@ -2,51 +2,69 @@
+By default, UI components have their accessible name (the programmatically associated label that assistive tech, like screen readers announce) computed from their own contents. That's part of the [spec'd algorithm](https://www.w3.org/TR/accname-1.1/) that browsers implement.
 
-## Lorem ipsum dolor sit amet.
+`aria-labelledby` can be used to point the algorithm somewhere else in the DOM. Or not.
 
-Lorem ipsum dolor sit amet `consectetur` adipisicing elit. In aperiam alias laborum magnam accusamus accusantium vero. Exercitationem distinctio at quisquam quibusdam rem voluptatum illo, laboriosam natus aperiam sequi, voluptate inventore!
+What happens if `aria-labelledby` points within?
 
+## Purgatory
 
 ```html
-<!DOCTYPE html>
-<html lang="en">
-  <head>
-	  <meta charset="utf-8" />
-	  <title>Webpage starter</title>
-  </head>
-  <body>
-    <h1>Hello World!</h1>
-  </body>
-</html>
+<a href="/updates" aria-labelledby="contents">
+  <span id="contents">Latest updates</span>
+</a>
 ```
 
-## Lorem ipsum dolor sit amet.
+## Testing
 
-* Lorem, ipsum.
-* Lorem, ipsum.
-* Lorem, ipsum.
-* Lorem, ipsum.
+The above snippet should be equivalent to `<a href="/updates"><span>Latest updates</span></a>`. Are there any issues besides being unnecessarily verbose code (and breaking the [first rule of ARIA](https://www.w3.org/TR/using-aria/#rule1))?
 
-![image](NSvIE.jpg)
+Technically, the [WAI-ARIA spec](https://www.w3.org/TR/wai-aria/#aria-labelledby) does not limit the DOM location of the element you point to.
 
-or
+I also checked how each major browser computes the element's accessible name via their developer tools (for example, [Chrome's DevTools](https://developer.chrome.com/docs/devtools/accessibility/reference#computed)):
 
-<img src="NSvIE.jpg" width="800" height="161" loading="lazy">
+- Chrome (129) ✅
+- Edge (129) ✅
+- Firefox (131) ✅
+- Safari (18.1) ✅
 
-Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsum, eligendi nobis impedit perferendis, saepe neque ullam laborum laudantium [recusandae]() expedita distinctio. At accusamus necessitatibus mollitia, quas officia incidunt nisi corrupti.
\ No newline at end of file
+I did only test those browsers on macOS, and I didn't test with different assistive technologies. But I trust that if the browser's tools show an accessible name of `"Latest updates"`, screen readers and voice control will have that value too.
+
+## Judgement
+
+As is, I'd feel like I was nitpicking by requesting the unnecessary `aria-labelledby` to be removed.
+
+But imagine this is in a large codebase with many developers working on it. Consider how easy it would be for someone tasked with adding a notification badge for example to the link to miss the implications of `aria-labelledby`:
+
+```html
+<a href="/updates" aria-labelledby="contents">
+  <span id="contents">Latest updates</span>
+  <span class="notification-badge">{{ updates.length }}</span>
```

Comment:
Is there a way to escape the `{{` so it renders?

Reply:
@matuzo if you get a chance. The curly braces don’t render to HTML as is

Reply:
Fixed in the advent24 branch